test/sys/test_wait.rs

```diff
@@ -12,7 +12,7 @@ fn test_wait_signal() {
 
     // Safe: The child only calls `pause` and/or `_exit`, which are async-signal-safe.
     match fork().expect("Error: Fork Failed") {
-      Child => pause().unwrap_or_else(|_| unsafe { _exit(123) }),
+      Child => pause(),
```

Comment:
I believe that you have to call `_exit()` instead of letting the thread naturally exit because of how Rust's standard cleanup isn't signal safe or something. @asomers Can you verify this?

Reply:
Yep.  The standard cleanup may try to access shared resources that were exclusively held by other threads at the time that `fork` happened.  But those other threads don't exist in the child, so your thread will never get the resources, and deadlock will result.  That's why we call `_exit` in child threads.

Reply:
Maybe I'm misunderstanding something but, isn't `unwrap_or_else` only called if `pause` fails and returns an error. Since that can't happend anymore `_exit` can never be called, and would have never be called even without the change.

Reply:
I've fixed this in #836, for some reason this pull request is now "from unknown repository". I don't know how to fix that, so instead I created an new pr.